abel/tools/analytical.py

```diff
@@ -250,59 +208,177 @@ def __init__(self, n, r_max, sigma=1.0, A0=1.0,
                           (np.abs(self.r) > 0)
 
 
-def abel_step_analytical(r, A0, r0, r1):
-    """
-    Directed Abel transform of a step function located between r0 and r1,
-    with a height A0
-
-    ::
-
-        A0 +                  +-------------+
-           |                  |             |
-           |                  |             |
-         0 | -----------------+             +-------------
-           +------------------+-------------+------------>
-           0                  r0            r1           r axis
-
-    This function is mostly used for unit testing the inverse Abel transform
-
-    Parameters
-    ----------
-    r1 : 1D array,
-        vecor of positions along the r axis. Must start with 0.
-    r0, r1 : floats,
-        positions of the step along the r axis
-    A0 : float or 1D array:
-        height of the step. If 1D array the height can be variable
-        along the Z axis
-
-    Returns
-    -------
-        1D array if A0 is a float, a 2D array otherwise
-    """
-
-    if np.all(r[[0, -1]]):
-        raise ValueError('The vector of r coordinates must start with 0.0')
-
-    F_1d = np.zeros(r.shape)
-    mask = (r >= r0)*(r < r1)
-    F_1d[mask] = 2*np.sqrt(r1**2 - r[mask]**2)
-    mask = r < r0
-    F_1d[mask] = 2*np.sqrt(r1**2 - r[mask]**2) - 2*np.sqrt(r0**2 - r[mask]**2)
-    A0 = np.atleast_1d(A0)
-    if A0.ndim == 1:
-        A0 = A0[:, np.newaxis]
-    return F_1d[np.newaxis, :]*A0
-
-
-def sym_abel_step_1d(r, A0, r0, r1):
-    """
-    Produces a symmetrical analytical transform of a 1d step
-    """
-    A0 = np.atleast_1d(A0)
-    d = np.empty(A0.shape + r.shape)
-    A0 = A0[np.newaxis, :]
-    for sens, mask in enumerate([r >= 0, r <= 0]):
-        d[:, mask] = abel_step_analytical(np.abs(r[mask]), A0, r0, r1)
-
-    return d
+class TransformPair(BaseAnalytical):
+
+    def __init__(self, n, profile=5):
+        """ Analytical Abel transform pairs: profile1, ..., profile7.
+            Table 1 of Chan and Hieftje Spectrochimica Acta B 61, 31-41 (2006)
+            do:10.1016/j.sab.2005.11.009
+
+            profile8:
+            curve B Hansen and Law JOSA 2, 510 (1985)
+            doi:10.1364/JOSAA.2.000510
+
+        Parameters
+        ----------
+        n : int
+            number of points along the r axis
+
+        profile: int
+            the profile number 1-8, see 'abel/tools/transform_pairs.py'
```

Comment:
note the profile 4 is broken.

Reply:
Done. However, the source function has only a slight discontinuity, and so in principle we may generate and least-squares fit our own projection function.